hashin.py

```diff
@@ -388,17 +388,39 @@ def amend_requirements_content(requirements, all_new_lines):
     padding = " " * 4
 
     def is_different_lines(old_lines, new_lines, indent):
+        # This regex is used to only temporarily normalize the names of packages
+        # in the lines being compared. This results in "old" names matching
+        # "new" names so that hashin correctly replaces them when it looks for
+        # them.
+        rex = re.compile("[-_]")
+
         # This assumes that the package is already mentioned in the old
         # requirements. Now we just need to double-check that its lines are
         # different.
         # The 'new_lines` is what we might intend to replace it with.
-        old = set([l.strip(" \\") for l in old_lines])
+        old = set([rex.sub("-", l.strip(" \\")) for l in old_lines])
```

Comment:
This will fail newer versions of `flake8` because `l` is a bad variable name. It's not your fault but do you mind
changing this to:
```suggestion
        old = set([rex.sub("-", line.strip(" \\")) for line in old_lines])
```
whilst you're at it.